app/classifier/components/Intervention/Intervention.jsx

```diff
@@ -14,10 +14,15 @@ const StyledInterventionMessage = styled.div`
   }
 `;
 
-function Intervention({ intervention, user }) {
+function Intervention({ onUnmount, intervention, user }) {
   const { message } = intervention;
   const checkbox = React.createRef();
 
+  useEffect(() => {
+    // the return value of an effect will be called to clean up after the component
+    return onUnmount;
```

Comment:
My reading suggests this will only run on unmount but it looks like effects will cleanup up if props change as well - https://reactjs.org/docs/hooks-effect.html#explanation-why-effects-run-on-each-update

I don't think this is an issue as we don't pass props to this function, thoughts?

Reply:
Good catch. The props here are `onUnmount, intervention, user` and we probably do want cleanup to run if any of those change.

Reply:
This might be useful.
>If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.
https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

Reply:
I've been playing around with this a bit and I think the only time it might happen is if the `intervention` prop changes while the component is being displayed. Is there a case where I might get a new intervention message while I'm already reading one?

Reply:
Yes - though in practice it would be slim. They don't send many messages but I could receive one while i'm viewing one, especially if the network / gateway has issues, i.e. received message at gateway, send to sugar but timeout to client who retries. 

Would the current code unmount and then update with the newly received message? Should we drop incoming messages until there are no interventions in the store?

I'd say no to switching the store to a list of messages, unless the incoming message has a validity period attached to it, https://github.com/zooniverse/interventions-gateway/issues/22

Reply:
My reading of the code is the classifier would receive a new `intervention` prop, which it would pass down to the `Intervention` component, which would then run `useEffect()` because of the props change? I'm not sure what would happen then: maybe the new intervention would render briefly, then be cleared by the cleanup code.

Reply:
I'm happy to add the validity period to the incoming messages via the gateway API if it helps simplify this issue.

Reply:
We could use the second argument to `useEffect` to ignore changes in the `intervention` prop. I'm not sure what the implications of that are, but it sounds like a case that shouldn't occur anyway.

At the moment, if a second message were received while you're currently reading one, it would overwrite it and the screen would update to show the new message.